src/main/java/frc/robot/Robot.java

```diff
@@ -291,37 +293,43 @@ private void spinControlPanel() {
         }
       }
 
-      String colorString;
+      String detectedColorString;
       Color detectedColor = colorSensor.getColor();
       ColorMatchResult match = colorMatcher.matchClosestColor(detectedColor);
       if (match.color == kBlueTarget) {
-        colorString = "Blue";
+        detectedColorString = "Blue";
       } else if (match.color == kRedTarget) {
-        colorString = "Red";
+        detectedColorString = "Red";
       } else if (match.color == kGreenTarget) {
-        colorString = "Green";
+        detectedColorString = "Green";
       } else if (match.color == kYellowTarget) {
-        colorString = "Yellow";
+        detectedColorString = "Yellow";
       } else {
-        colorString = "Unknown";
+        detectedColorString = "Unknown";
       }
-      detectedColorEntry.setString(colorString);
+      String lastDetectedColorString = detectedColorEntry.toString(); 
+      detectedColorEntry.setString(detectedColorString);
       confidenceEntry.setDouble(match.confidence);
-
+      
       int doubleContolPanelSpinAmount = controlPanelSpinAmount * 2;
-      if (targetControlPanelColor == "Green" && (colorString != "Green" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Red" && (colorString != "Red" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Blue" && (colorString != "Blue" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Yellow" && (colorString != "Yellow" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      }
+      turnControlPanel(detectedColorString, targetControlPanelColor, lastDetectedColorString,
+          doubleContolPanelSpinAmount);
       targetSpinEntry.setDouble(controlPanelSpinAmount);
     } else {
       detectedColorEntry.setString("N/A");
       confidenceEntry.setDouble(0);
     }
   }
+
+  /*
+   * This function turns the control panel when called upon in spinControlPanel().
+   */
+  public void turnControlPanel(String detectedColor, String targetColor, String lastDetectedColor, int spinAmount) {
+    if (targetColor != detectedColor || spinAmount > 0) {
+      controlPanelTalon.set(controlPanelSpinSpeed);
+    }
+    if (targetColor == detectedColor && lastDetectedColor != detectedColor && spinAmount > 0) {
+      spinAmount -= 1;
```

Comment:
This will only decrement a copy of  `spinAmount`. I feel that, instead of this, the spin amount member variable should be decremented in the other method, after the `detectedColorString == lastDetectedColorString` mentioned in an above comment.

Reply:
The most recent commit should resolve this.